File: Code/Websites/DanpheEMR/wwwroot/DanpheApp/src/app/clinical/eye-examination/shared/SmileSettings.model.ts
import {
  NgForm,
  FormGroup,
  FormControl,
  Validators,
  FormBuilder,
  ReactiveFormsModule
} from '@angular/forms';
import * as moment from 'moment/moment';

export class SmileSettingsModel {
  public Id: number = 0;
  public MasterId: number = 0;
  public SpotDistanceLent: string = null;
  public SpotDistanceLentSide: string = null;
  public SpotDistanceCap: string = null;
  public SpotDistanceCapSide: string = null;
  public TrackDistanceLent: string = null;
  public TrackDistanceLentSide: string = null;
  public TrackDistanceCap: string = null;
  public TrackDistanceCapSide: string = null;
  public EnergyOffsetLent: string = null;
  public EnergyOffsetLentSide: string = null;
  public EnergyOffsetCap: string = null;
  public EnergyOffsetCapSide: string = null;
  public ScanDirectionLent: string = null;
  public ScanDirectionLentSide: string = null;
  public ScanDirectionCap: string = null;
  public ScanDirectionCapSide: string = null;
  public ScanModeLent: string = null;
  public ScanModeLentSide: string = null;
  public ScanModeCap: string = null;
  public ScanModeCapSide: string = null;
  public MinThicknessLent: string = null;
  public MinThicknessLentSide: string = null;
  public MinThicknessCap: string = null;
  public MinThicknessCapSide: string = null;
  public SidecutLent: string = null;
  public SidecutLentSide: string = null;
  public SidecutCap: string = null;
  public SidecutCapSide: string = null;
  public CreatedBy: number = 0;
  public CreatedOn: Date = new Date();
  public IsOD: boolean;

  constructor() {

  }
  dateValidator(control: FormControl): { [key: string]: boolean } {
    var currDate = moment().format('YYYY-MM-DD HH:mm');
    if (control.value) { // gets empty string for invalid date such as 30th Feb or 31st Nov)
      if ((moment(control.value).diff(currDate) > 0)
        || (moment(currDate).diff(control.value, 'years') > 200)) //can select date upto 200 year past from today.
        return { 'wrongDate': true };
    }
    else
      return { 'wrongDate': true };
  }

}

File: Code/Components/DanpheEMR.ServerModel/ClinicalModels/ClinicalEye/SmileSettingsModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DanpheEMR.ServerModel
{
    public class SMILESSettingsModel
    {
        [Key]
        public int Id { get; set; }
        public int MasterId { get; set; }
        public string SpotDistanceLent { get; set; }
        public string SpotDistanceLentSide { get; set; }
        public string SpotDistanceCap { get; set; }
        public string SpotDistanceCapSide { get; set; }
        public string TrackDistanceLent { get; set; }
        public string TrackDistanceLentSide { get; set; }
        public string TrackDistanceCap { get; set; }
        public string TrackDistanceCapSide { get; set; }
        public string EnergyOffsetLent { get; set; }
        public string EnergyOffsetLentSide { get; set; }
        public string EnergyOffsetCap { get; set; }
        public string EnergyOffsetCapSide { get; set; }
        public string ScanDirectionLent { get; set; }
        public string ScanDirectionLentSide { get; set; }
        public string ScanDirectionCap { get; set; }
        public string ScanDirectionCapSide { get; set; }
        public string ScanModeLent { get; set; }
        public string ScanModeLentSide { get; set; }
        public string ScanModeCap { get; set; }
        public string ScanModeCapSide { get; set; }
        public string MinThicknessLent { get; set; }
        public string MinThicknessLentSide { get; set; }
        public string MinThicknessCap { get; set; }
        public string MinThicknessCapSide { get; set; }
        public string SidecutLent { get; set; }
        public string SidecutLentSide { get; set; }
        public string SidecutCap { get; set; }
        public string SidecutCapSide { get; set; }
        public int CreatedBy { get; set; }
        public DateTime CreatedOn { get; set; }
        public Boolean IsOD { get; set; }

    }
}


